packages/web-components/src/components/ic-chip/ic-chip.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  Component,
+  h,
+  Prop,
+  State,
+  Listen,
+  Event,
+  EventEmitter,
+  Element,
+} from "@stencil/core";
+import {
+  onComponentRequiredPropUndefined,
+  isSlotUsed,
+} from "../../utils/helpers";
+import { IcChipAppearance, IcChipSizes } from "./ic-chip.types";
+import dismissIcon from "../../assets/dismiss-icon.svg";
+
+/**
+ * @slot icon - Content will be rendered at the start of the chip.
+ */
+@Component({
+  tag: "ic-chip",
+  styleUrl: "ic-chip.css",
+  shadow: true,
+})
+export class Chip {
+  @Element() el: HTMLIcChipElement;
+
+  /**
+   * The text rendered within the chip.
+   */
+  @Prop() label!: string;
+  /**
+   * The emphasis of the chip.
+   */
+  @Prop() appearance?: IcChipAppearance = "filled";
+  /**
+   * The size of the chip.
+   */
+  @Prop() size?: IcChipSizes = "default";
+  /**
+   * If `true`, the chip will have a close button at the end to dismiss it.
+   */
+  @Prop() dismissible?: boolean = false;
+  /**
+   * If `true`, the chip will appear disabled.
+   */
+  @Prop() disabled?: boolean = false;
+
+  @State() visible: boolean = true;
+
+  @State() isHovered: boolean = false;
+
+  @Listen("dismiss", { capture: true })
+  handleClick(): void {
+    this.visible = !this.visible;
+  }
+  /**
+   * Is emitted when the user dismisses the chip.
+   */
+  @Event() dismiss: EventEmitter<void>;
+
+  private dismissAction = (): void => {
+    this.dismiss.emit();
+  };
+
+  private mouseEnterHandler = (): void => {
+    this.isHovered = true;
+  };
+
+  private mouseLeaveHandler = (): void => {
+    this.isHovered = false;
+  };
+
+  componentDidLoad(): void {
+    onComponentRequiredPropUndefined(
+      [{ prop: this.label, propName: "label" }],
+      "Chip"
+    );
+  }
+
+  render() {
+    const {
+      label,
+      appearance,
+      size,
+      dismissible,
+      visible,
+      disabled,
+      isHovered,
+    } = this;
+
+    return (
+      visible && (
+        <div
+          class={{
+            ["chip"]: true,
+            [`${appearance}`]: true,
+            [`${size}`]: true,
+            ["disabled"]: disabled,
+            ["dismissible"]: dismissible,
+            ["hovered"]: isHovered,
+          }}
+        >
+          {isSlotUsed(this.el, "icon") && (
+            <div class="icon">
+              <slot name="icon" />
+            </div>
+          )}
+          <ic-typography
+            variant="label"
+            apply-vertical-margins={false}
+            class="label"
+          >
+            <span>{label}</span>
```

Comment:
I'm thinking if is it worth passing through the label as a slot instead of a prop so it appears in the Light DOM? Just been testing the React variant of chips using React Testing library. I appreciate the dev can use `data-testid` but putting the label as a slot allows the dev to also use `getByText` as well as `getByTestId` so there is added flexibility.

Reply:
Yeah I'm not sure it'd be possible. @GCHQ-Developer-847 I'm not sure if you have ideas?
One of the other issues with having it as a slot would be that it'd be harder to restrict devs from not putting anything other than text in the chip, i.e we wouldn't want them to add a button etc.

Reply:
Yeah, I get what you mean. I think the onus should be on the developer to get that right (re: adding a button). We can have something in our guidance for it as well just to cover our backs. 

However, after further investigation, I have been able to get the following to work, so we should be okay.

```jsx
const { getByTestId } = render(<IcChip data-testid='chip' label='User' />)

const chip = getByTestId('chip');

await waitFor(() => {
  expect(chip).toHaveClass('hydrated');
})

expect(chip.label).toBe('User')
```

We can force dev in this direction if they have any issues.